internal/fs/handle/file.go

```diff
@@ -71,9 +71,13 @@ func NewFileHandle(inode *inode.FileInode, fileCacheHandler *file.CacheHandler,
 
 // Destroy any resources associated with the handle, which must not be used
 // again.
+// LOCK_FUNCTION(fh.inode.mu)
+// UNLOCK_FUNCTION(fh.inode.mu)
 func (fh *FileHandle) Destroy() {
 	// Deregister the fileHandle with the inode.
+	fh.inode.Lock()
```

Comment:
Trying to understand if locking is necessary here or not?

Reply:
Yes lock is required for DeRegisterFileHandle. This is a new method which we added in the last PR only, at that time lock was taken at fs level. Now i am lowering it to inode level.

Reply:
I couldnt see the fs level lock being removed in this commit. Was in a previous commit?

Reply:
it is in this PR only. in fs.go changes.